### PR TITLE
fix: added operand type to ConditionalStudioComponentProperty

### PR DESCRIPTION
**Pull request by kpranoto-aws**
*Description of changes:*
* Added operandType in ConditionalStudioComponentProperty
* Parsing string values of operand to match type with operandType
* Non-breaking change. Backwards-compatible with previous behavior (treated value as string)

### CODE REVIEW

File: packages/codegen-ui-react/lib/react-component-render-helper.ts [hunk 1]
```diff
@@ -324,19 +325,46 @@ export function getSyntaxKindToken(operator: RelationalOperator): BinaryOperator
...
     case 'number':
-      return factory.createNumericLiteral(operand);
+      return factory.createNumericLiteral(value);
     case 'boolean':
-      return operand ? factory.createTrue() : factory.createFalse();
+      return value ? factory.createTrue() : factory.createFalse();
+    case 'string':
+      return factory.createStringLiteral(value);
     default:
-      return factory.createStringLiteral(operand);
```
Comment (alharris-at) [inline packages/codegen-ui-react/lib/react-component-render-helper.ts:334]: I think I'd prefer to leave this behavior as-is for now. Is there a need to change this?

File: packages/codegen-ui-react/lib/react-component-render-helper.ts [hunk 2]
```diff
@@ -324,19 +325,46 @@ export function getSyntaxKindToken(operator: RelationalOperator): BinaryOperator
...
+  operand: string | number | boolean,
+  operandType: string | undefined,
+): Expression {
+  if (typeof operand === 'string' && operandType && operandType !== 'string') {
+    return valueToCorrectLiteral(operand, operandType);
+  }
+  return valueToJsxLiteral(operand);
+}
+
+function valueToCorrectLiteral(value: any, valueType: string): PrimaryExpression {
```
Comment (alharris-at) [inline packages/codegen-ui-react/lib/react-component-render-helper.ts:338]: perhaps rename to `typedValueToLiteral`?

File: packages/codegen-ui-react/lib/__tests__/react-component-render-helper.test.ts [hunk 3]
```diff
@@ -27,10 +27,36 @@ import {
   isDefaultValueOnly,
   getSyntaxKindToken,
   buildChildElement,
+  buildConditionalExpression,
 } from '../react-component-render-helper';
 
 import { assertASTMatchesSnapshot } from './__utils__/snapshot-helpers';
 
+Error.stackTraceLimit = Infinity;
```
[the remark this answers is not in the document]
Comment (kpranoto-aws) [reply packages/codegen-ui-react/lib/__tests__/react-component-render-helper.test.ts]: removed

File: packages/codegen-ui-react/lib/react-component-render-helper.ts (hunk 2 above)
Comment (kpranoto-aws) [reply packages/codegen-ui-react/lib/react-component-render-helper.ts]: fixed naming to distinguish between the two functions better

**Review by alharris-at — APPROVED** [review]

**Comment by alharris-at** [comment]
Looks good Kevin, once our functional tests complete I'll merge into `develop`.

**Review by alharris-at — APPROVED** [review]

**Pull request merged**